operator-framework/src/main/java/io/javaoperatorsdk/operator/api/Controller.java

```diff
@@ -1,21 +1,20 @@
 package io.javaoperatorsdk.operator.api;
 
-import io.fabric8.kubernetes.client.CustomResource;
+import io.quarkus.runtime.annotations.RegisterForReflection;
 
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 @Retention(RetentionPolicy.RUNTIME)
+@RegisterForReflection
```

Comment:
Is this the only reason we have quarkus on classpath? Would be cleaner if we could do it without it.

Reply:
Right, good point. 
We can create project that will act like "springboot-starter" for quarkus. It will take of usign supported quarkus version and configuration of reflections.

Reply:
Yep, I'm planning on doing that with #228.

Reply:
So my suggestion is to:
1) Remove quarkus from this PR
2) Open new PR dedicated only for implementing quarkus starter 
If @metacosm don't mind I can take care of this

Reply:
That said, I'm not sure that native compilation will work properly without using `RegisterForReflection`… I also think we should merge #229 as soon as possible because that will simplify a lot of things by removing the need to deal with `Doneable` classes.

Reply:
@kirek007 another aspect is also that #228 tries to extract the configuration parts from the Spring Boot starter so that it could be reused in a Quarkus extension instead of duplicating the code so I really think that it should get done sooner rather than later. That said, I'm not quite satisfied with how it currently is, in terms of configuration I mean.

Reply:
I can remove the Quarkus dependency from here, the other way to achieve the same is generating a ReflectionConfigurationFile in compile time (just like what Quarkus is doing with the annotations). Can be done in the new module that will be added.